lightning/src/ln/channelmanager.rs

```diff
@@ -6540,14 +6512,32 @@ where
 
 								true
 							},
-							ChannelPhase::UnfundedInboundV1(chan) => {
-								process_unfunded_channel_tick!(peer_state, chan, pending_msg_events)
-							},
-							ChannelPhase::UnfundedOutboundV1(chan) => {
-								process_unfunded_channel_tick!(peer_state, chan, pending_msg_events)
-							},
-							ChannelPhase::UnfundedV2(chan) => {
-								process_unfunded_channel_tick!(peer_state, chan, pending_msg_events)
+							None => {
+								phase.context_mut().maybe_expire_prev_config();
+								let unfunded_context = phase.unfunded_context_mut().expect("channel should be unfunded");
```

Comment:
Nice that we got rid of the macro, but sad that we need this `expect`. That's okay, though.

Reply:
Yeah, the only way to really avoid this is to have a method returning an enum that has `Funded` and `Unfunded` variants, where the latter gives the `UnfundedChannelContext`. I'm not necessarily opposed to the idea (and maybe it would be useful elsewhere?), but I opted for a minimal change for now.